lib/ansible/modules/notification/office_365_connector_card.py

```diff
@@ -0,0 +1,291 @@
+        section_payload['activityTitle'] = section['activity_title']
+
+    if 'activity_subtitle' in section:
+        section_payload['activitySubtitle'] = section['activity_subtitle']
+
+    if 'activity_text' in section:
+        section_payload['activityText'] = section['activity_text']
+
+    if 'hero_image' in section:
+        section_payload['heroImage'] = section['hero_image']
+
+    if 'text' in section:
+        section_payload['text'] = section['text']
+
+    if 'facts' in section:
+        section_payload['facts'] = section['facts']
+
+    if 'images' in section:
+        section_payload['images'] = section['images']
+
+    if 'actions' in section:
+        section_payload['potentialAction'] = build_actions(section['actions'])
+
+    return section_payload
+
+
+def build_payload_for_connector_card(module, summary, color, title, text, actions, sections):
+    payload = dict()
+    payload['@context'] = OFFICE_365_CARD_CONTEXT
+    payload['@type'] = OFFICE_365_CARD_TYPE
+
+    if summary is not None:
+        payload['summary'] = summary
+
+    if color is not None:
+        payload['themeColor'] = color
+
+    if title is not None:
+        payload['title'] = title
+
+    if text is not None:
+        payload['text'] = text
+
+    if actions:
+        payload['potentialAction'] = build_actions(actions)
+
+    if sections:
+        payload['sections'] = build_sections(sections)
+
+    payload = module.jsonify(payload)
+    return payload
+
+
+def do_notify_connector_card_webhook(module, webhook, payload):
+    headers = {
+        'Content-Type': 'application/json'
+    }
+
+    response, info = fetch_url(
+        module=module,
+        url=webhook,
+        headers=headers,
+        method='POST',
+        data=payload
+    )
+
+    if info['status'] == 200:
+        module.exit_json(changed=True)
+    else:
+        module.fail_json(
+            msg="failed to send %s as a connector card to Incoming Webhook: %s"
+                % (payload, info['msg'])
+        )
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            webhook=dict(required=True, no_log=True),
+            summary=dict(),
+            color=dict(),
+            title=dict(),
+            text=dict(),
+            actions=dict(type='list'),
+            sections=dict(type='list')
+        ),
+        supports_check_mode=True
+    )
+
+    webhook = module.params['webhook']
+    summary = module.params['summary']
+    color = module.params['color']
+    title = module.params['title']
+    text = module.params['text']
+    actions = module.params['actions']
+    sections = module.params['sections']
+
+    payload = build_payload_for_connector_card(
+        module,
+        summary,
+        color,
+        title,
+        text,
+        actions,
+        sections)
+
+    if module.check_mode:
+        # In check mode, exit before actually sending the message
+        module.exit_json(changed=True, msg=payload)
```

Comment:
I prefer that the module would check the connection itself as well, without actually sending the message. If the API supports stub messages (or empty messages?), use that. Otherwise just test the authentication/connection some other way.